xtask/src/main.rs

```diff
@@ -0,0 +1,113 @@
+use anyhow::{Context, Result};
+use clap::{Parser, Subcommand, ValueEnum};
+use std::{env, path::PathBuf, process::Command};
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Build WASM package
+    BuildWasm {
+        /// Output directory for wasm-pack
+        #[arg(long, required = true)]
+        out_dir: String,
+
+        /// Target platform (nodejs or web)
+        #[arg(long, required = true, value_enum)]
+        target: Target,
+    },
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, ValueEnum)]
+enum Target {
+    /// Build for NodeJS
+    Nodejs,
+    /// Build for Web
+    Web,
+}
+
+impl From<Target> for &'static str {
+    fn from(target: Target) -> &'static str {
+        match target {
+            Target::Nodejs => "nodejs",
+            Target::Web => "web",
+        }
+    }
+}
+
+fn main() -> Result<()> {
+    let cli = Cli::parse();
+
+    match &cli.command {
+        Commands::BuildWasm { out_dir, target } => build_wasm(out_dir, *target),
+    }
+}
+
+fn build_wasm(out_dir: &str, target: Target) -> Result<()> {
+    const RUSTFLAGS: &str = "-C target-feature=+atomics,+bulk-memory,+mutable-globals -C link-arg=--max-memory=4294967296";
```

Comment:
I guess [this](https://github.com/o1-labs/proof-systems/blob/master/Makefile#L34) can be removed too then?